labgrid/remote/exporter.py

```diff
@@ -273,6 +274,79 @@ def _stop(self, start_params):
 exports["USBSerialPort"] = SerialPortExport
 exports["RawSerialPort"] = SerialPortExport
 
+
+@attr.s(eq=False)
+class VivadoHWServerExport(ResourceExport):
+    """ResourceExport for a XilinxUSBJTAG via Vivado Hardware Server"""
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self.data['cls'] = "NetworkXilinxUSBJTAG"
+        from ..resource.udev import XilinxUSBJTAG
+        self.local = XilinxUSBJTAG(target=None, name=None, **self.local_params)
+        self.child = None
+
+    def __del__(self):
+        if self.child is not None:
+            self.stop()
+
+    def _get_serial(self):
+        if self.local.serial:
+            return self.local.serial
+        else:
+            return self.local.serial_short
+
+    def _get_params(self):
+        """Helper function to return parameters"""
+        return {
+            'host': self.host,
+            'serial': self._get_serial(),
+            'hw_server_cmd': self.local.hw_server_cmd,
+            'agent_url': self.local.agent_url,
+            'gdb_port': self.local.gdb_port,
+            'log_level': self.local.log_level,
+            'extra_args': self.local.extra_args
+        }
+
+    def _start(self, start_params):
+        """Start ``hw_server`` subprocess"""
+        assert self.local.avail
+
+        if not self.local.agent_url:
+            self.local.agent_url = "tcp::" + str(get_free_port())
+
+        if not self.local.gdb_port:
+            # TODO: Get a range of 4 consecutive free ports
+            self.local.gdb_port = get_free_port()
+
+        args = [
+            self.local.hw_server_cmd,
+            '-e', "'set jtag-port-filter {}'".format(self._get_serial()),
+            '-s{}'.format(self.local.agent_url),
+            '-p{}'.format(self.local.gdb_port),
+        ]
+        if self.local.log_level:
+            args.append('-l{}'.format(','.join(self.local.log_level)))
+        args.extend(self.local.extra_args)
+        # TODO: Add timeout
+        self.child = subprocess.Popen(' '.join(args), preexec_fn=os.setsid,
```

Comment:
Please use `start_new_session=True` instead of `preexec_fn=os.setsid`.

Reply:
Done